index.bs

```diff
@@ -180,88 +181,180 @@ public IP address scope is being discussed in
+the [=UA=] shall meet the following requirements:
+- The device identifier (domain name) should be easy to remember and short enough but
+    not to recognize as fraudulent by anti-phishing.
+- The [=device=] and the [=UA=] should have a pairing method by using the device
+    interface (e.g., pushing a specific button on the [=device=] to start the pairing,
+    showing the PIN code for the pairing on the [=device=]’s display).
+
+### Alerting User and User's Approval ### {#req-alerting-user}
+
+The [=web service=] loaded on the [=UA=] should not be able to access the [=device=]
+without an explicit approval from the user especially during its first time access.
+This is essential in cases where the [=device=] is not capable of using a [=Web PKI certificate=].
+In such cases, the [=UA=] will possibly use an alternative identifier (e.g., [=non-Web PKI certificate=], PSK, RPK)
+but it must not be trusted without the user’s approval.
+Therefore, considering the [=UA=] shall meet the following requirement:
+- When the [=web service=] requests the [=UA=] to access the [=device=] in local network,
+    the [=UA=] shall show a popup window to obtain the user’s approval and shall permit
+    the access only if the user approve the access.
+- When the [=device=] does not use a [=Web PKI certificate=] for HTTPS/WSS communication,
+    the [=UA=] shall have a way to obtain user’s approval to trust an alternative identifier
+    (e.g., [=non-Web PKI certificate=], PSK, RPK).
+
+### Certificate Management ### {#req-certificate-management}
+
+When the [=device=] obtains a TLS server certificate for HTTPS/WSS communication,
+the [=CA=], [=UA=] and the [=device=] shall meet the following requirements:
+- The [=CA=] that issues certificates for [=device=]s shall be able to revoke the certificates.
+- The [=UA=] shall be able to detect the revocation of the [=device=]’s certificate.
+- The [=device=] shall be able to refresh the certificate at any time.
+
+### <dfn>Device Discovery</dfn> ### {#req-device-discovery}
+
+As described in [[#uc-02]] and [[#uc-03]], when the [=web service=] does not know the device URL in advance,
+the [=device=] needs to be discovered. That means the discovery process must meet the following requirements:
+- The [=UA=] shall be able to discover the presence of [=device=]s in the same local network.
+- The [=UA=] shall be able to obtain the endpoint URL of the [=device=] that has the scheme `https://` or `wss://`.
+- When the [=UA=] finds multiple [=device=]s, the [=UA=] shall be able to present the selector interface
+    to display the list of the [=device=]s and allow the user to select one specific [=device=].
+
+## Non-Functional Requirements ## {#req-non-funcational-req}
+
+The non-functional requirements are:
+- Security
+- Privacy
+- Availability
+- Scalability
+- User Experience
+
+### Security ### {#req-security}
+
+Following security considerations are important:
```

Comment:
s/Following security considerations are important/It is essential to take care of the following security considerations/